actionpack/lib/action_controller/test_case.rb

```diff
@@ -578,10 +580,19 @@ def setup_request(controller_class_name, action, parameters, session, flash, xhr
           end
         end
 
+        def wrap_execution(&block)
+          if executor_around_each_request && defined?(Rails.application) && Rails.application
+            Rails.application.executor.perform(&block)
```

Comment:
```suggestion
            Rails.application.executor.wrap(&block)
```

I believe you want `wrap` here, as to simulate the behavior of `ActionDispatch::Executor`. `perform` is only meant for wrapping the entire test as to allow one more level of `wrap`. But here you won't want to allow any more reset after this one.